src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -649,14 +649,37 @@ public static IEnumerable<String> FilterCompatibleSources(Architecture chosenPla
         }
 
         /// <summary>
-        /// Returns true if source settings are incomaptible with target settings.
+        /// Returns true if target framework is incompatible with any source framework, that is run needs to be aborted
+        /// </summary>
+        public static bool IsFrameworkIncompatible(Framework chosenFramework, IDictionary<String, Framework> sourceFrameworks)
+        {
+            bool isSettingIncompatible = false;
+            foreach (var source in sourceFrameworks.Keys)
+            {
+                Framework actualFramework = sourceFrameworks[source];
+                string sourceFrameworkName = actualFramework.Name.Split(',')[0];
+                string targetFrameworkName = chosenFramework.Name.Split(',')[0];
+
+                if (sourceFrameworkName.Equals(".NETFramework", StringComparison.OrdinalIgnoreCase) && targetFrameworkName.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase) ||
+                   sourceFrameworkName.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase) && targetFrameworkName.Equals(".NETFramework", StringComparison.OrdinalIgnoreCase))
+                {
+                    isSettingIncompatible = true;
+                }
+            }
+
+            return isSettingIncompatible;
+        }
+
+        /// <summary>
+        /// Returns true if source settings are incompatible with target settings.
         /// </summary>
         private static bool IsSettingIncompatible(Architecture sourcePlatform,
-            Architecture targetPlatform,
-            Framework sourceFramework,
-            Framework targetFramework)
+        Architecture targetPlatform,
+        Framework sourceFramework,
+        Framework targetFramework)
         {
-            return IsPlatformIncompatible(sourcePlatform, targetPlatform) || IsFrameworkIncompatible(sourceFramework, targetFramework);
+            var res =  IsPlatformIncompatible(sourcePlatform, targetPlatform) || IsFrameworkIncompatible(sourceFramework, targetFramework);
```

Comment:
nit: no change needed

Reply:
Restructured this code. `IsSettingIncompatible `is directly inferring platform/frameworks incompatibility without calling anything else